src/CodeGen_Xtensa_vectors.template.cpp

```diff
@@ -1164,7 +1164,7 @@ template<>
 HALIDE_ALWAYS_INLINE void store_narrowing<native_vector_i16, int8_t, VECTOR_WIDTH_I16>(const native_vector_i16 &a, void *base, int32_t offset) {
     valign align = IVP_ZALIGN();
     xb_vecNx8 *__restrict ptr = (xb_vecNx8 *)((int8_t *)base + offset);
-    IVP_SANX8S_IP(a, align, ptr);
+    IVP_SANX8S_IP((a << 8) >> 8, align, ptr);
```

Comment:
I am not sure I understand this part, could you please add comment?

Reply:
overflow in xtensa and x86 behaves differently. This move makes sure that we truncate and not saturate